packages/react-components/react-tree/stories/src/Tree/TreeNavigationModeTreeGrid.stories.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { Tree, TreeItem, TreeItemLayout, TreeItemProps } from '@fluentui/react-components';
+import { Edit20Regular, MoreHorizontal20Regular } from '@fluentui/react-icons';
+import {
+  Button,
+  Menu,
+  MenuItem,
+  MenuList,
+  MenuPopover,
+  MenuTrigger,
+  useRestoreFocusTarget,
+} from '@fluentui/react-components';
+
+type CustomTreeItemProps = TreeItemProps;
+
+const CustomTreeItem = ({ children, ...props }: CustomTreeItemProps) => {
+  const focusTargetAttribute = useRestoreFocusTarget();
+  const [layoutChildren, subtree] = React.Children.toArray(children);
+
+  // same items to be used between contextmenu and actions
+  const commonMenuItems = (
+    <>
+      <MenuItem>New </MenuItem>
+      <MenuItem>New Window</MenuItem>
+      <MenuItem disabled>Open File</MenuItem>
+      <MenuItem>Open Folder</MenuItem>
+    </>
+  );
+
+  return (
+    <Menu positioning="below-end" openOnContext>
+      <MenuTrigger disableButtonEnhancement>
+        <TreeItem aria-description="has actions" {...focusTargetAttribute} {...props}>
+          <TreeItemLayout
+            actions={
+              <>
+                <Button aria-label="Edit" appearance="subtle" icon={<Edit20Regular />} />
+                <Menu>
+                  <MenuTrigger disableButtonEnhancement>
+                    <Button aria-label="More options" appearance="subtle" icon={<MoreHorizontal20Regular />} />
+                  </MenuTrigger>
+                  <MenuPopover>
+                    <MenuList>{commonMenuItems}</MenuList>
+                  </MenuPopover>
+                </Menu>
+              </>
+            }
+          >
+            {layoutChildren}
+          </TreeItemLayout>
+          {subtree}
+        </TreeItem>
+      </MenuTrigger>
+      <MenuPopover>
+        <MenuList>
+          <MenuItem>Edit</MenuItem>
+          {commonMenuItems}
+        </MenuList>
+      </MenuPopover>
+    </Menu>
+  );
+};
+
+export const NavigationModeTreeGrid = () => {
```

Comment:
Pressing down arrow from the actions should actually navigation to the next tree item

Reply:
done 😁